hub/core/chunk.py

```diff
@@ -108,6 +114,30 @@ def update_headers(self, incoming_num_bytes: int, sample_shape: Tuple[int]):
         self.shapes_encoder.register_samples(sample_shape, 1)
         self.byte_positions_encoder.register_samples(num_bytes_per_sample, 1)
 
+    def update_sample(
+        self, local_sample_index: int, new_buffer: memoryview, new_shape: Tuple[int]
+    ):
+        """Updates data and headers for `local_sample_index` with the incoming `new_buffer` and `new_shape`."""
+
+        expected_dimensionality = len(self.shapes_encoder[local_sample_index])
+        if expected_dimensionality != len(new_shape):
+            raise TensorInvalidSampleShapeError(new_shape, expected_dimensionality)
+
+        self._make_data_bytearray()
+
+        sb, eb = self.byte_positions_encoder[local_sample_index]
+        new_nb = len(new_buffer)
+
+        # update data
+        # TODO: optimize this (lots of memcps)
```

Comment:
should be easy to do right now, just create a bigger bytearray and assign left, new_buffer, and right to it. We should do this before merging